src/pages/Swap/index.tsx

```diff
@@ -438,74 +442,44 @@ export default function Swap({ history }: RouteComponentProps) {
-                            <Trans>
-                              <HideSmall>Back to </HideSmall>
-                              V3
-                            </Trans>
-                          </TYPE.main>
-                        </ButtonGray>
-                      )
-                    ))}
-
-                  {toggledVersion === Version.v3 && trade && isTradeBetter(v2Trade, v3Trade) && (
-                    <ButtonGray
-                      width="fit-content"
-                      padding="0.1rem 0.5rem"
-                      disabled
-                      style={{
-                        display: 'flex',
-                        justifyContent: 'space-between',
-                        height: '24px',
-                        opacity: 0.8,
-                        marginLeft: '0.25rem',
-                      }}
-                    >
-                      <TYPE.black fontSize={12}>
-                        <Trans>V3</Trans>
-                      </TYPE.black>
-                    </ButtonGray>
-                  )}
+            {!showWrap && trade && (
+              <Row justify={!trade ? 'center' : 'space-between'}>
+                <RowFixed style={{ position: 'relative' }}>
+                  <MouseoverTooltipContent
+                    Container={({ children }) => <ResponsiveTooltipContainer>{children}</ResponsiveTooltipContainer>}
+                    content={<SwapRoute trade={trade} syncing={routeIsSyncing} />}
+                    placement="bottom"
+                  >
+                    <AutoRow gap="4px" width="auto">
+                      <AutoRouterLogo />
+                      {trade instanceof V3Trade && trade.swaps.length > 1 && (
+                        <TYPE.blue fontSize={14}>{trade.swaps.length}</TYPE.blue>
+                      )}
+                    </AutoRow>
+                  </MouseoverTooltipContent>
                 </RowFixed>
-                {trade ? (
-                  <RowFixed>
+                <RowFixed>
+                  <LoadingOpacityContainer $loading={routeIsSyncing}>
                     <TradePrice
                       price={trade.executionPrice}
                       showInverted={showInverted}
                       setShowInverted={setShowInverted}
                     />
-                    <MouseoverTooltipContent
-                      content={<AdvancedSwapDetails trade={trade} allowedSlippage={allowedSlippage} />}
-                    >
-                      <StyledInfo />
-                    </MouseoverTooltipContent>
-                  </RowFixed>
-                ) : null}
+                  </LoadingOpacityContainer>
+                  <MouseoverTooltipContent
+                    Container={({ children }) => (
+                      <ResponsiveTooltipContainer origin="bottom right" width={'295px'}>
+                        {children}
+                      </ResponsiveTooltipContainer>
+                    )}
```

Comment:
this inline container component does not need to be defined in every render cycle